src/apis/platformvm/api.ts

```diff
@@ -473,7 +474,32 @@ export class PlatformVMAPI extends JRPCAPI {
       "platform.getBalance",
       params
     )
-    return response.data.result
+
+    const result = response.data.result
+
+    const parseDict = (input: any[]): BalanceDict => {
+      var dict: BalanceDict = {}
+      for (const [k, v] of Object.entries(input)) dict[k] = new BN(v)
+      return dict
+    }
+
+    if (this.core.getNetwork().P.lockModeBondDeposit) {
+      return {
+        balances: parseDict(result.balances),
+        unlockedOutputs: parseDict(result.unlockedOutputs),
+        bondedOutputs: parseDict(result.bondedOutputs),
+        depositedOutputs: parseDict(result.depositedOutputs),
+        bondedDepositedOutputs: parseDict(result.bondedDepositedOutputs),
+        utxoIDs: result.utxoIDs
+      } as GetBalanceResponse
+    }
+    return {
+      balance: new BN(result.balance),
+      unlocked: new BN(result.unlocked),
+      lockedStakeable: new BN(result.lockedStakeable),
+      lockedNotStakeable: new BN(result.lockedNotStakeable),
+      utxoIDs: result.utxoIDs
+    } as GetBalanceResponse
```

Comment:
Just to be sure - should both returns be in GetBalanceResponse? No GetBalanceResponseAvax here?

Reply:
The compiler was happy with this, at the end of the day you have to choose what interface you want to use.